cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -438,3 +457,21 @@ func validateCertificateWithConfig(cert *x509.Certificate, baseName string, cfg
 	}
 	return nil
 }
+
+// CheckCertificatePeriodValidity takes a certificate and prints a warning if its period
+// is not valid related to the current time. It does so only if the certificate was not validated already
+// by keeping track with a cache.
+func CheckCertificatePeriodValidity(baseName string, cert *x509.Certificate) {
```

Comment:
i'm still debating whether the cache is wanted here. the only benefit is that during a "kubeadm" process execution, it can omit printing the same validation warning multiple times for the same certificate.